SCF-Session-8-Assignment/src/question/Zoo.java

```diff
@@ -0,0 +1,195 @@
+package question;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is the main ZOO management class which contains all the methods to add and remove animal, cage and zone.
+ * @author Jugal
+ * Dated 07/24/2019
+ */
+public class Zoo {
+
+	List<Zone> zoneList = new ArrayList<>();
+	
+	/**
+	 * This method creates a new zone in the Zoo
+	 * @param zoneCategory is the category of the zone to be created
+	 * @param ZoneCapacity is the capacity of the zone to be created
+	 */
+	public void addZone(String zoneCategory, int ZoneCapacity)
+	{
+		zoneList.add(new Zone(zoneCategory, ZoneCapacity, true,true));
+	}
+	
+	/**
+	 * This method creates a new cage in the zone
+	 * @param cageType is the type of the cage to be created
+	 * @param cageCapacity is the capacity of the cage to be created
+	 */
+	public void addCage(String cageType, int cageCapacity)
+	{
+		int i;
+		for(i = 0; i < zoneList.size(); i++)
+		{
+			if(zoneList.get(i).zoneCategory.equals(getCategory(cageType)) && zoneList.get(i).zoneCapacity > zoneList.get(i).cageList.size())
+			{
+				zoneList.get(i).cageList.add(new Cage(cageType, cageCapacity));
+				break;
+			}
+		}
+		
+		assert i == zoneList.size() : "Cage cannot be created";
+	}
+	
+	/**
+	 * This is the helper method which calculates the category of the animal type
+	 * @param type is the type of animal whose category is to be calculated
+	 * @return {String} category of the animal type
+	 */
+	String getCategory(String type)
+	{
+		String category = null;
+		
+		switch(type.toUpperCase())
+		{
+			case "LION":
+				category = Lion.class.getSuperclass().getSimpleName();
+				break;
+			case "PEACOCK":
+				category = Peacock.class.getSuperclass().getSimpleName();
+				break;
+			case "SNAKE":
+				category = Snake.class.getSuperclass().getSimpleName();
+				break;
+		}
+		return category;
+	}
+	
+	/**
+	 * This is the helper method used to build the Animal type object
+	 * @param animalName is the animal name whose object is to be build
+	 * @param animalType is the type of that animal name
+	 * @return {Animal} animal object
+	 */
+	public Animal buildAnimalObject(String animalName, String animalType)
+	{
+		Animal animalObj = null;
+		switch (animalType.toLowerCase())
+		{
+		case "lion":
+			animalObj = new Lion(animalName, animalType, Lion.class.getSuperclass().getSimpleName());
+			break;
+		
+		case "peacock":
+			animalObj = new Peacock(animalName, animalType, Peacock.class.getSuperclass().getSimpleName());

```

Comment:
get the category using getCatgory() method you declared.

Reply:
Done.
Changed the logic.